fiftyone/core/map/process.py

```diff
@@ -0,0 +1,302 @@
+                        # the various workers to be yielded first.
+                        else:
+                            sample_errors.append((sample_id, err, None))
+
+                    else:
+                        # Yield successfully mapped sample
+                        yield sample_id, None, result
+
+            queue.close()
+            queue.join_thread()
+
+            # It is possible to aggregate one error per worker. There
+            # might be a better way to handle this in the future but for
+            # now, return the first error seen.
+            if sample_errors:
+                yield sample_errors[0]
+
+
+def _init_worker(
+    dataset_name: str,
+    view_stages: Any,
+    map_fcn: bytes,
+    batch_count: Optional[multiprocessing.Value],  # type: ignore
+    sample_count: Optional[multiprocessing.Value],  # type: ignore
+    queue: Optional[multiprocessing.Queue],
+    cancel_event: multiprocessing.Event,  # type: ignore
+    save: bool,
+    progress: bool,
+    skip_failures: bool,
+    lock: Optional[multiprocessing.RLock],  # type: ignore
+):
+    # pylint:disable=import-outside-toplevel
+    # pylint:disable=reimported
+    # pylint:disable=redefined-outer-name
+    from tqdm.auto import tqdm
+
+    import fiftyone as fo
+    import fiftyone.core.odm.database as food
+    import fiftyone.core.view as fov
+
+    # pylint:disable=global-variable-undefined
+    global process_sample_collection
+    global process_map_fcn
+    global process_batch_count
+    global process_sample_count
+    global process_queue
+    global process_cancel_event
+    global process_save
+    global process_progress
+    global process_skip_failures
+
+    # Ensure that each process creates its own MongoDB clients
+    # https://pymongo.readthedocs.io/en/stable/faq.html#using-pymongo-with-multiprocessing
+    # pylint:disable-next=protected-access
+    food._disconnect()
+
+    dataset = fo.load_dataset(dataset_name)
+    if view_stages:
+        # pylint:disable-next=protected-access
+        process_sample_collection = fov.DatasetView._build(
+            dataset, view_stages
+        )
+    else:
+        process_sample_collection = dataset
+
+    process_map_fcn = pickle.loads(map_fcn)
+    process_batch_count = batch_count
+    process_sample_count = sample_count
+    process_queue = queue
+    process_cancel_event = cancel_event
+    process_save = save
+    process_progress = progress
+    process_skip_failures = skip_failures
+
+    if lock is not None:
+        tqdm.set_lock(lock)
+
+
+def _map_batch(args: Tuple[int, int, fomb.SampleBatch]):
+    i, num_batches, batch = args
+
+    try:
+        sample_collection = batch.create_subset(process_sample_collection)
+
+        sample_iter = sample_collection.iter_samples(autosave=process_save)
+
+        pb = None
+        if process_progress:
+            desc = f"Batch {i:0{len(str(num_batches))}}/{num_batches}"
```

Comment:
ah interesting. could use a util method or comment, this tripped me up for a second